src/admin-events/admin-events.js

```diff
@@ -64,38 +60,67 @@ function createAlert(eventName, contractName, contractAddress, eventType, eventS
       contractName,
       contractAddress,
       eventName,
+      strippedArgs,
     },
   });
 }
 
-async function handleTransaction(txEvent) {
+// Populates the contracts array
+async function initialize() {
+  // Constant for getAddress
+  const CHAIN_ID = 1;
+
+  // Clear the contracts array
+  contracts.length = 0;
+
+  // Get the contract names that have events we wish to monitor
+  const contractNames = Object.keys(adminEvents);
+
+  // Get and store the information about each contract
+  await Promise.all(contractNames.map(async (name) => {
+    const abi = getAbi(name);
+    const iface = new ethers.utils.Interface(abi);
+    const address = (await getAddress(name, CHAIN_ID)).toLowerCase();
+
+    const contract = {
+      name,
+      address,
+      iface,
+    };
+    contracts.push(contract);
+  }));
+}
```

Comment:
we have a slightly different style for using `initialize`. I refactored it to fit the way we plan on using it on `AE-516_initializaton_refactor`, check it out and if you like it you can totally merge it into this branch :)